### PR TITLE
feat: forbid using attr names that are ambiguous with $nested.syntax$

### PR DESCRIPTION
**Pull request by ephys**
### Pull Request Checklist

- [x] Have you added new tests to prevent regressions?
- [x] Does `npm run test` or `npm run test-DIALECT` pass with this change (including linting)?
- [x] Is a documentation update included (if this change modifies existing APIs, or introduces new ones)?
- [n/a] Did you update the typescript typings accordingly (if applicable)?
- [n/a] Does the description below contain a link to an existing issue (Closes #[issue]) or a description of the issue you are solving?
- [x] Did you follow the commit message conventions explained in [CONTRIBUTING.md](https://github.com/sequelize/sequelize/blob/main/CONTRIBUTING.md)?

### Description Of Change

v7 only

It's currently possible to name an attribute `$name$` or `$na.me$`. This is ambiguous with the `$nested.syntax$` that can be used in queries.

This PR makes that syntax reserved and rejects any attribute named in a way that would be ambiguous with the nested attribute reference syntax.

It's opinionated as I found a couple places in the code where Sequelize *tries* to make it work, but there are many, many other places where it just silently breaks.

This is a breaking change, and should be marked as such when squashing the commits from this PR.

### CODE REVIEW

**Comment by ephys** [comment]
I think we should forbid using `.` in attribute names too, as that would be ambiguous with the `json.reference.syntax` and the `$nested.attribute.reference.syntax$`

File: docs/manual/other-topics/upgrade-to-v7.md [hunk 1]
```diff
@@ -16,6 +16,18 @@ As a result, the manual typings that were formerly best-effort guesses on top of
 have been removed and all typings are now directly retrieved from the actual TypeScript code.
 You'll likely find many tiny differences which however should be easy to fix.
 
+## Attributes cannot start and end with `$`, or include `.`
```
Comment (WikiRik) [inline docs/manual/other-topics/upgrade-to-v7.md:19]: Maybe good to add an example of how people can change the names of the attributes they are using?

Comment (ephys) [reply docs/manual/other-topics/upgrade-to-v7.md]: Done! :)

**Review by WikiRik — APPROVED** [review]

**Comment by sdepold** [comment]
Can we add this to the list of breaking changes in the migration docs?

**Comment by ephys** [comment]
It is :)

**Comment by sdepold** [comment]
Ah true :) Guess I didn't see it on my phone. Let's get this into main then!

**Pull request merged**